plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -288,12 +294,32 @@ getCardanoTxOutRefs = onCardanoTx txOutRefs cardanoApiTxOutRefs
 getCardanoTxOutputs :: CardanoTx -> [TxOut]
 getCardanoTxOutputs = fmap fst . getCardanoTxOutRefs
 
-getCardanoTxUnspentOutputsTx :: CardanoTx -> Map V1.Tx.TxOutRef TxOut
-getCardanoTxUnspentOutputsTx = Map.fromList . fmap swap . getCardanoTxOutRefs
+getCardanoTxProducedOutputs :: CardanoTx -> Map V1.Tx.TxOutRef TxOut
+getCardanoTxProducedOutputs = Map.fromList . fmap swap . getCardanoTxOutRefs
 
 getCardanoTxSpentOutputs :: CardanoTx -> Set V1.Tx.TxOutRef
 getCardanoTxSpentOutputs = Set.fromList . map txInRef . getCardanoTxInputs
 
+getCardanoTxReturnCollateral :: CardanoTx -> Maybe TxOut
+getCardanoTxReturnCollateral = onCardanoTx txReturnCollateral
+    (\(CardanoApiEmulatorEraTx (C.Tx (C.TxBody C.TxBodyContent {..}) _)) ->
+        case txReturnCollateral of
+            C.TxReturnCollateralNone     -> Nothing
+            C.TxReturnCollateral _ txOut -> Just $ TxOut txOut
+    )
+
+getCardanoTxProducedCollateralOutputs :: CardanoTx -> Map V1.Tx.TxOutRef TxOut
```

Comment:
Is it useful? There's only ever one return collateral. Maybe simply rename to singular and return `(TxOutRef, TxOut)`.

Also, we should use the official term "return collateral" instead of "collateral output"

Reply:
It is used as an alternative to `getCardanoTxProducedOutputs` when the transaction is phase 2 invalid. So it should have the same type signature.